### PR TITLE
Can now update config file through the web ui

### PR DESCRIPTION
**Pull request by chenshuiluke**
Changes made to settings will be written to the config.yaml file

### CODE REVIEW

File: infoset/api/config_page.py [hunk 1]
```diff
@@ -20,4 +24,91 @@ def getConfig():
 
     """
     # Return
-    return render_template('config.html')
+    configDict = get_current_config_file_contents()
+    return render_template('config.html', config=configDict)
+
+@CONFIG_PAGE.route('/config/update',methods=['POST'])
+def update_config():
+    print("HERE")
```
Comment (Pr0x1m4) [inline infoset/api/config_page.py]: Remove print statement

File: infoset/api/config_page.py [hunk 2]
```diff
@@ -20,4 +24,91 @@ def getConfig():
...
+    configDict['log-level'] = config.log_level()
+
+    configDict['memcached-host'] = config.memcached_hostname()
+    configDict['memcached-port'] = config.memcached_port()
+
+    configDict['sqlalchemy-overflow'] = config.sqlalchemy_max_overflow()
+    configDict['sqlalchemy-pool-size'] = config.sqlalchemy_pool_size()
+    return configDict
+
+def associate_config_form_field_names_with_settings_names():
```
Comment (Pr0x1m4) [inline infoset/api/config_page.py]: Variable and function names are waay too long, use shorter names
Eg. 
```python
def _getConfig
    config = {
    ...
    }
    return config
```

File: infoset/api/config_page.py [hunk 3]
```diff
@@ -20,4 +24,91 @@ def getConfig():
...
+    dirname, filename = os.path.split(os.path.abspath(__file__))
+    config_file_path = dirname + '/../../etc/config.yaml'
+    config_file = open(config_file_path, 'w')
+
+    yaml_text = yaml.dump(new_yaml_obj, config_file,default_flow_style=False)
+    #yaml_text = yaml.dump(new_yaml_obj, default_flow_style=False)
+
+    return get_config()
+
+def get_current_config_file_contents():
```
Comment (Pr0x1m4) [inline infoset/api/config_page.py]: Prefix helper methods like this one with an underscore and make function names shorter `_get_current_config()`

File: infoset/api/config_page.py [hunk 4]
```diff
@@ -20,4 +24,91 @@ def getConfig():
...
+            pass
+        new_yaml_obj['main'][setting_name] = value
+
+    
+    dirname, filename = os.path.split(os.path.abspath(__file__))
+    config_file_path = dirname + '/../../etc/config.yaml'
+    config_file = open(config_file_path, 'w')
+
+    yaml_text = yaml.dump(new_yaml_obj, config_file,default_flow_style=False)
+    #yaml_text = yaml.dump(new_yaml_obj, default_flow_style=False)
```
Comment (Pr0x1m4) [inline infoset/api/config_page.py]: Remove commented line

**Comment by chenshuiluke** [comment]
Made some of the requested adjustments

**Pull request merged**